Counterexamples/OrderedCancelAddCommMonoidWithBounds.lean

```diff
@@ -0,0 +1,31 @@
+/-
+Copyright (c) 2023 Martin Dvorak. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Martin Dvorak
+-/
+import Mathlib.Algebra.Order.Monoid.Defs
+import Mathlib.Order.CompleteLattice
+
+/-!
+# OrderedCancelAddCommMonoid + BoundedOrder = bad idea
+
+This file shows that combining `OrderedCancelAddCommMonoid` with `BoundedOrder` is not a good idea,
+as it forbids any strict inequalities (`x < y`).
+The same applies to any superclasses, e.g. combining
+`StrictOrderedSemiring` with `CompleteLattice`.
+The crux is that cancellation properties don't like the `⊥` and `⊤` elements.
+-/
+
+private class BoundedOrderedCancelAddCommMonoid (α : Type*)
+  extends OrderedCancelAddCommMonoid α, BoundedOrder α
+
+example {α : Type*} [BoundedOrderedCancelAddCommMonoid α] :
+    (∃ x y : α, x < y) → False := by
+  rintro ⟨x, y, hxy⟩
```

Comment:
No need for the new class (or the exists):
```suggestion
example {α : Type*} [OrderedCancelAddCommMonoid α] [BoundedOrder α] (x y : α) :
    ¬(x < y) := by
  intro hxy
```

Reply:
How can I be sure that `[OrderedCancelAddCommMonoid α] [BoundedOrder α]` refer to the same order relation?

Reply:
Because `BoundedOrder` takes `[LE α]` as an argument.